music21/metadata/__init__.py

```diff
@@ -683,80 +1267,15 @@ def alternativeTitle(self):
         >>> md.alternativeTitle
         'Heroic Symphony'
         '''
-        result = self._workIds['alternativeTitle']
-        if result is not None:
-            return str(result)
-
-    @alternativeTitle.setter
-    def alternativeTitle(self, value):
-        self._workIds['alternativeTitle'] = Text(value)
-
-    def _contributor_role_getter(self, role: str) -> t.Optional[str]:
-        '''
-        get the name of the first contributor with this role, or None
-
-        see composer property for an example
-        '''
-        result = self.getContributorsByRole(role)
-        if result is not None:
-            # get just the name of the first contributor
-            return str(result[0].name)
-        return None
-
-    def _contributor_role_setter(self, role: str, name: str) -> None:
-        '''
-        set the name for a particular role, adding a new contributor
-        in the process if none exists for that role.
-
-        see composer.setter property for an example
-        '''
-        c = None
-
-        result = self.getContributorsByRole(role)
-        if result is not None:
-            c = result[0]
-        else:
-            c = Contributor()
-            c.role = role
-            self.contributors.append(c)
-
-        c.name = name
-
-    def _contributor_multiple_role_getter(self, role: str) -> t.List[str]:
-        '''
-        get a list of the names of contributors with a certain role.
-
-        see composers (plural) property for an example.
-        '''
-        result = self.getContributorsByRole(role)
-        if result is not None:
-            # get just the name of each composer/role
-            return [x.name for x in result]
-        else:
-            return []
-
-    def _contributor_multiple_role_setter(self, role: str, value: t.List[str]) -> None:
-        '''
-        set multiple names for a particular role, replacing the people
-        already in those roles.
-
-        see composers.setter (plural) property for an example.
-        '''
-        existing_contributors = self.getContributorsByRole(role)
-        if existing_contributors:
-            for existing in existing_contributors:
-                self.contributors.remove(existing)
-
-        for v in value:
-            contrib = Contributor(role=role, name=v)
-            self.addContributor(contrib)
+        return self._getSingularAttribute('alternativeTitle')
 
     @property
     def composer(self):
         r'''
         Get or set the composer of this work. Only the first composer can be
-        got or set via properties.  Instead add Composer roles to the .contributors
-        list.
+        got or set via properties.  You can add multiple composers by setting
+        them dictionary-style (e.g. md['composer'] = ['composer1', 'composer2']),
+        or by using md.add('composer', ...) API.
```

Comment:
Better to give the example of how to do the latter after the basic example has been shown.